src/request.js

```diff
@@ -136,4 +137,27 @@ module.exports = class Request extends EventEmitter {
     }
     return null;
   }
+
+  // Temporarily suspends the flow of data from the database.
+  // No more 'row' events will be emitted until resume() is called.
+  pause() {
+    if (this.paused) {
+      return;
+    }
+    this.paused = true;
```

Comment:
The paused state of the request objects is not the same as the paused state of the token stream transform.
See above comment for the SENT_CLIENT_REQUEST exit event.